api/hooks/index.js

```diff
@@ -31,25 +40,75 @@ module.exports = sails => {
       const options = sails.config.rabbitworker.options
 
       // default retry connection attempts to 3
-      if (!_.has(options.retryConnectionAttempts)) {
+      if (!_.has(options, 'retryConnectionAttempts')) {
         options.retryConnectionAttempts = 3
       }
 
-      if (!_.has(options.retryDroppedConnectionAttempts)) {
+      if (!_.has(options, 'retryDroppedConnectionAttempts')) {
         options.retryDroppedConnectionAttempts = 100
       }
 
       // default retry connection timeout to 10 seconds
-      if (!_.has(options.retryConnectionTimeout)) {
+      if (!_.has(options, 'retryConnectionTimeout')) {
         options.retryConnectionTimeout = 10000
       }
+
+      // default retry failed jobs
+      if (!_.has(options, 'retryFailedJobs')) {
+        options.retryFailedJobs = true
+      }
+
+
+
       connect(options)
+        .catch(err => {
+          return handleDroppedConnection(err, options)
+        })
         .then(() => next())
         .catch(next)
     }
   }
 }
 
+function setupChannel(conn, options, jobs) {
+  // set up the channel and the delayed job exchange
+  return conn.createChannel().then(function(ch) {
+    const exchangeName = options.exchangeName || 'sails_jobs'
+    sails.rabbitworker.channel = ch
+
+    // log channel level errors
+    ch.on('error', err => {
+      sails.log.error(err)
+    })
+
+    ch.on('close', () => {
+      sails.log.verbose('Rabbitmq channel closed')
+    })
+
+    // create a durable 'direct' exchange, to pass jobs to queues based on exact name match
+    return ch.assertExchange(exchangeName, 'direct', {
+      durable: true
+    }).then(() => {
+      // create a method that accepts the job name and the message data and publishes it to the appropriate queue
+      sails.createJob = (queueName, payload, options) => {
```

Comment:
This looks a little funky, but I don't necessarily have a better proposal off the top of my head.